infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -411,23 +463,26 @@ private void handleFlushCacheRequest(CorfuMsg msg, ChannelHandlerContext ctx, IS
      */
     @Override
     public void sealServerWithEpoch(long epoch) {
-        CorfuPayloadMsg<Long> msg = new CorfuPayloadMsg<>();
-        msg.setEpoch(epoch);
-        msg.setPriorityLevel(PriorityLevel.HIGH);
+        RequestMsg batchProcessorMsg = getRequestMsg(
+                getHighPriorityHeaderMsg(HeaderMsg.newBuilder().setEpoch(epoch).build()),
+                getSealRequestMsg(epoch)
+        );
+
         try {
-            batchWriter.addTask(SEAL, msg).join();
+            batchWriter.addTask(BatchWriterOperation.Type.SEAL, batchProcessorMsg).join();
```

Comment:
We should have a uniform way to propagate these requests. For example `sealServerWithEpoch` should be invoked with a seal message instead of only a long.